test/hotspot/jtreg/compiler/vectorapi/TestSelectFromTwoVectorOp.java

```diff
@@ -0,0 +1,453 @@
+        for (int i = 0; i < SPECIES.loopBound(ba.length); i += SPECIES.length()) {
+            ByteVector.fromArray(SPECIES, bindex, i)
+                .selectFrom(ByteVector.fromArray(SPECIES, ba, i),
+                            ByteVector.fromArray(SPECIES, bb, i))
+                .intoArray(bres, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_8, ">0"},
+        applyIfCPUFeature = {"asimd", "true"})
+    public static void selectFromTwoVector_Byte64() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_64, ba, bb, bindex[0]);
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_16, ">0"},
+        applyIfCPUFeature = {"asimd", "true"})
+    public static void selectFromTwoVector_Byte128() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_128, ba, bb, bindex[1]);
+    }
+
+    @Test
+    @IR(failOn = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_32},
+        applyIfCPUFeatureAnd = {"asimd", "true", "sve2", "false"},
+        applyIf = {"MaxVectorSize", ">=32"})
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_32, ">0"},
+        applyIfCPUFeature = {"sve2", "true"},
+        applyIf = {"MaxVectorSize", ">=32"})
+    public static void selectFromTwoVector_Byte256() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_256, ba, bb, bindex[2]);
+    }
+
+    @Test
+    @IR(failOn = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_64},
+        applyIfCPUFeatureAnd = {"asimd", "true", "sve2", "false"},
+        applyIf = {"MaxVectorSize", ">=64"})
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VB, IRNode.VECTOR_SIZE_64, ">0"},
+        applyIfCPUFeature = {"sve2", "true"},
+        applyIf = {"MaxVectorSize", ">=64"})
+    public static void selectFromTwoVector_Byte512() {
+        ByteSelectFromTwoVectorKernel(ByteVector.SPECIES_512, ba, bb, bindex[3]);
+    }
+
+    // Test SelectFromTwoVector operation for Shorts
+    @ForceInline
+    public static void ShortSelectFromTwoVectorKernel(VectorSpecies SPECIES, short[] sa,
+                                                      short[] sb, short[] sindex) {
+        for (int i = 0; i < SPECIES.loopBound(sa.length); i += SPECIES.length()) {
+            ShortVector.fromArray(SPECIES, sindex, i)
+                .selectFrom(ShortVector.fromArray(SPECIES, sa, i),
+                            ShortVector.fromArray(SPECIES, sb, i))
+                .intoArray(sres, i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VS, IRNode.VECTOR_SIZE_4, ">0"},
+        applyIfCPUFeature = {"asimd", "true"})
+    public static void selectFromTwoVector_Short64() {
+        ShortSelectFromTwoVectorKernel(ShortVector.SPECIES_64, sa, sb, sindex[0]);
+    }
+
+    @Test
+    @IR(counts = {IRNode.SELECT_FROM_TWO_VECTOR_VS, IRNode.VECTOR_SIZE_8, ">0"},
```

Comment:
Hi @Bhavana-Kilambi ,
Kindly also include x86-specific feature checks in IR rule for this test.

You can directly integrate attached patch.

[select_from_ir_feature.txt](https://github.com/user-attachments/files/21034639/select_from_ir_feature.txt)

Reply:
Thank you @jatin-bhateja . Will do that in my next patch with other changes.

Reply:
Hi @jatin-bhateja , have you tested `jdk/incubator/vector` tests with your patch?